Formula/macvim.rb

```diff
@@ -42,6 +34,13 @@ def install
     # make sure that CC is set to "clang"
     ENV.clang
 
+    # set `arch` appropriately
+    arch = if Hardware::CPU.arm? && !Hardware::CPU.in_rosetta2?
+      "arm64"
+    else
+      "x86_64"
+    end
```

Comment:
@Homebrew/maintainers don't we have a method for formulae to get this easier?

Reply:
`Hardware::CPU.arch`?

Reply:
`Hardware::CPU.arch` maybe? 
I don't have an Apple Silicon to check that it does exactly the same

Reply:
If there is, it doesn't look like it's in `Hardware::CPU`.

Reply:
I think `Hardware::CPU.arch` calls [this](
https://github.com/Homebrew/brew/blob/b0d2a64b31706f6d90da9c967058a8adfa2f4eb6/Library/Homebrew/hardware.rb#L56-L60):
```
      def arch_64_bit
        if arm?
          :arm64
        elsif intel?
          :x86_64
```
so it doesn't look like it's what you want.

Reply:
Wouldn't calling `to_s` on those give you the same result?

Reply:
Yes, it should. Sorry, I don't really grok Ruby. 😅

Reply:
Does this handle Rosetta2 like the above?

Reply:
It does handle Rosetta. It's defined [here](https://github.com/Homebrew/brew/blob/b0d2a64b31706f6d90da9c967058a8adfa2f4eb6/Library/Homebrew/hardware.rb#L86-L93):

```ruby
      def type
        case RUBY_PLATFORM
        when /x86_64/, /i\d86/ then :intel
        when /arm/, /aarch64/ then :arm
        when /ppc|powerpc/ then :ppc
        else :dunno
        end
      end
```

`RUBY_PLATFORM` is going to return the right thing. In general I would imagine very few formulas need to use `in_rosetta2?` because Rosetta already does all the hard work of pretending to be an x86_64 environment for you. You would only need to detect it if you are doing something off-nominal, and you would be calling `in_rosetta2?` when the arch is x86_64, not when it's arm64. I think if you just call `arch.to_s` that should be all you need.

Reply:
Will do, then!